src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/ITagHelperComponentContextInitializer.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// Provides methods to initialize <see cref="ViewContext"/> for <see cref="ITagHelperComponent"/>s.
+    /// </summary>
+    public interface ITagHelperComponentContextInitializer
+    {
```

Comment:
Open to suggestions for names

Reply:
Left a comment on the implementation class so it was easily referred to by another comment I left there 😄

Reply:
Is `ITagHelperComponentActivator` in use?

Reply:
&hellip; or `ITagHelperComponentPropertyActivator` since we use `Activator` alone mostly for things that create container instances.